install.sh

```diff
@@ -845,9 +868,16 @@ do_install() {
     systemctl daemon-reload >/dev/null
     info "Enabling rancher-system-agent.service"
     systemctl enable rancher-system-agent
+
+    if [ ${TRANSACTIONAL_UPDATE} ] && [ -x /usr/sbin/transactional-update ]; then
+        info "Rebooting to envoke rancher-system-agent on transactional-update server."
+        systemctl reboot -f
```

Comment:
The brazen nature of this reboot is going to be a problem -- we run this install script from two potential places, either the standard shell for a user, or via a system-upgrade-controller job. In the prior case, I would say that we should provide some type of "opt out" or cancellation prompt (if interactive) to prevent the script from suddenly rebooting the user's machine, albeit they probably would expect it. 

On the secondary issue, we do not have granular control over when the system-upgrade-controller job runs. This means that on a Rancher upgrade, a user will potentially see their entire downstream cluster nodes reboot, with no rhyme or reason to it (we do not drain nodes for our SUC plan as of now, as this script is intended to be non-invasive and should not cause downtime). Level setting for this expectation of behavior will need to be done, before we can merge a change like this.

Reply:
Yeah, I ignorantly thought that this script only executed once at installation time and rebooting a node wouldn't be a big deal. Hmm, on the first issue, do you think a "ctrl-c" type timeout would be preferable to an environment variable?

On the second, okay, I wasn't aware that we also run this script on a system-upgrade-controller job. We are only rebooting because of placement of `/usr/bin/rancher-system-agent` (and the uninstaller). I take it these files are also replaced during an upgrade with more recent versions if available? Or do they remain the same? If they remain the same, we could check for the existence of those files and skip the reboot, as they wouldn't be visible if they had just been installed (with transactional-update at least).